test/http_tiles.cc

```diff
@@ -258,6 +261,64 @@ TEST(HttpTiles, test_graphreader_multiple_threads) {
   test_graphreader_tile_download(8, 2, 4);
 }
 
+TEST(HttpTiles, test_interrupt) {
+  using namespace baldr;
+
+  TestTileDownloadData params;
+  const auto url_builder = [&params](size_t index) -> std::string {
+    auto test_tile_index = index % params.test_tile_names.size();
+    auto tile_name = params.test_tile_names[test_tile_index];
+    auto tile_uri = params.tile_url_base;
+    tile_uri += tile_name;
+    tile_uri += params.request_params;
+    return tile_uri;
+  };
+
+  const auto non_existent_tile_id = params.get_nonexistent_tile_id();
+  std::unordered_set<std::string> canceled_uris{url_builder(0), url_builder(2)};
+
+  curl_tile_getter_t tile_getter(2, "", params.is_gzipped_tile);
+  const curl_tile_getter_t::interrupt_t interrupt = [&canceled_uris](const auto& uri) {
+    return canceled_uris.find(uri) != canceled_uris.end();
+  };
+
+  tile_getter.set_interrupt(&interrupt);
+  const auto thread_count = 2;
+  std::vector<std::thread> threads;
+  threads.reserve(thread_count);
+  for (int thread_i = 0; thread_i < thread_count; ++thread_i) {
+    threads.emplace_back([&, thread_i]() {
+      for (int tile_i = 0; tile_i < params.test_tile_ids.size(); ++tile_i) {
+        bool is_for_this_thread = ((tile_i % thread_count) == thread_i);
+        if (!is_for_this_thread) {
+          continue;
+        }
+
+        auto test_tile_index = tile_i % params.test_tile_names.size();
+        auto expected_tile_id = params.test_tile_ids[test_tile_index];
+
+        auto tile_uri = url_builder(tile_i);
+        if (canceled_uris.find(tile_uri) != canceled_uris.end()) {
+          EXPECT_THROW(tile_getter.get(tile_uri), std::runtime_error);
```

Comment:
in this case it seems to throw before event starting network activity?

Reply:
not really, we will have first throw only after first progress is received.